libs/git_wrapper/Git_wrapper.ml

```diff
@@ -158,6 +158,20 @@ let get_git_root_path () =
   | Ok (path, (_, `Exited 0)) -> path
   | _ -> raise (Error "Could not get git root from git rev-parse")
 
+(* this does not technically use any Git features, because JSCaml cannot always
+   successfully call get_git_root_path. but, it's a convenient place to put
+   it.
+*)
+let chdir_to_repo_root () =
+  let rec parent changed =
+    if Sys.getcwd () = "/" then invalid_arg "couldn't find repo root"
+    else if not (Sys.file_exists ".git" && Sys.is_directory ".git") then (
```

Comment:
this is dangerous because if there is no .git, then after calling this function you will have
modified the cwd by side effect! 
You should at least finalize and restore the cwd.

But I think it would be better instead to have a pure function called

find_git_root_path_in_parents
that does not do any chdir!